sv_pipeline/load_data.py

```diff
@@ -0,0 +1,392 @@
+    variant_id = get_variant_id(row, header_indices)
+    if variant_id not in parsed_svs_by_id:
+        parsed_svs_by_id[variant_id] = get_parsed_column_values(row, header_indices, CORE_COLUMNS)
+        parsed_svs_by_id[variant_id][COL_CONFIGS[VAR_NAME_COL]['field_name']] = variant_id
+        parsed_svs_by_id[variant_id][GENOTYPES_FIELD] = []
+
+    sample_info = get_parsed_column_values(row, header_indices, SAMPLE_COLUMNS)
+    sample_info[SAMPLE_ID_FIELD] = sample_id
+
+    sv = parsed_svs_by_id[variant_id]
+    sv[GENOTYPES_FIELD].append(sample_info)
+    # Use the largest coordinates for the merged SV
+    sv[START_COL] = min(sv.get(START_COL, float('inf')), sample_info[START_COL])
+    sv[END_COL] = max(sv.get(END_COL, 0), sample_info[END_COL])
+    sv[NUM_EXON_FIELD] = max(sv.get(NUM_EXON_FIELD, 0), sample_info[NUM_EXON_FIELD])
+
+
+def load_file(file_path, parse_row, out_file_path=None, columns=None):
+    out_file = None
+    if out_file_path:
+        out_file = open(out_file_path, 'w')
+
+    with open(file_path, 'r') as f:
+        header = f.readline()
+        header_indices = {col: i for i, col in enumerate(header.split())}
+        missing_cols = [col for col in columns or COLUMNS if col not in header_indices]
+        if missing_cols:
+            raise Exception('Missing expected columns: {}'.format(', '.join(missing_cols)))
+        if out_file:
+            out_file.write(header)
+
+        for line in tqdm(f, unit=' rows'):
+            row = line.split()
+            parsed = parse_row(row, header_indices)
+            if parsed and out_file:
+                out_file.write(line)
+
+    if out_file:
+        out_file.close()
+
+
+def subset_and_group_svs(input_dataset, sample_subset, sample_remap, ignore_missing_samples, write_subsetted_bed=False):
+    parsed_svs_by_name = {}
+    found_samples = set()
+    skipped_samples = set()
+    out_file_path = None
+    if write_subsetted_bed:
+        file_path = input_dataset.split('/')
+        file_path[-1] = 'subset_{}'.format(file_path[-1])
+        out_file_path ='/'.join(file_path)
+
+    def _parse_row(row, header_indices):
+        sample_id = get_field_val(row, SAMPLE_COL, header_indices)
+        if sample_remap and sample_id in sample_remap:
+            sample_id = sample_remap[sample_id]
+        if sample_subset is None or sample_id in sample_subset:
+            parse_sv_row(row, parsed_svs_by_name, header_indices, sample_id)
+            found_samples.add(sample_id)
+            return True
+        else:
+            skipped_samples.add(sample_id)
+            return False
+
+    load_file(input_dataset, _parse_row, out_file_path=out_file_path)
+
+    print('Found {} sample ids'.format(len(found_samples)))
+    if sample_subset:
+        if len(found_samples) != len(sample_subset):
+            missed_samples = sample_subset - found_samples
+            missing_sample_error = 'Missing the following {} samples:\n{}'.format(
```

Comment:
what happens if you find more samples than are in the sample subset?

Reply:
we usually do! We are loading from a file with all the CMG samples and we are only looking for one project's worth. We just don't load those